### PR TITLE
Make the bearer token visible in FlightSqlServiceClient

### LINKED ISSUE #6253: Make the bearer token visible in FlightSqlServiceClient
`FlightSqlServiceClient` stores the bearer token it receives on the Handshake response and uses it on all subsequent requests. The Flight protocol does not clearly specify whether the bearer token has to be sent on DoGet calls as well (along with the ticket). Because of this, several Flight SQL server implementors decided to expect the bearer token on DoGet requests.
However, if the `DoGet` call needs to go to a different endpoint, or when the client decides to parallelise the streams and have the `DoGet` calls performed by different threads (or even nodes), the new client instances cannot access the bearer token simply because the field is private and only writable (set_ and clear_).

Adding a "getter" would solve this problem and allow the client app to propagate the token to distributed DoGet calls.

### PR DESCRIPTION
**Pull request by ccciudatu**
# Which issue does this PR close?

Closes #6253.

# Rationale for this change
 

Allow FlightSQL clients to reuse the bearer token received on handshake for distributed DoGet calls.

# What changes are included in this PR?

A mere "getter" :)

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-flight/src/sql/client.rs [hunk 1]
```diff
@@ -110,6 +110,11 @@ impl FlightSqlServiceClient<Channel> {
         self.token = None;
     }
 
+    /// Share the bearer token with potentially different `DoGet` clients
+    pub fn token(&self) -> &Option<String> {
+        &self.token
+    }
```
Comment (alamb) [inline arrow-flight/src/sql/client.rs:116]: I think a more "ideomatic" API would return an Option, like:

```suggestion
    pub fn token(&self) -> Option<&String> {
        self.token.as_ref()
    }
```

**Review by alamb — APPROVED** [review]
Makes sense to me -- thank you @ccciudatu 

I have a suggestion for your consideration but I don't think it is required

**Comment by ccciudatu** [comment]
> I have a suggestion for your consideration but I don't think it is required

@alamb ack, thanks!

**Comment by alamb** [comment]
Thanks again @ccciudatu

**Pull request merged**